src/modules/ipfs/ipfs.service.ts

```diff
@@ -26,4 +54,48 @@ export class IPFSService {
       return false;
     }
   }
+
+  /**
+   * Get claim from cluster. If claim isn't found tries to get from gateway
+   *
+   * @param cid Content identifier.
+   * @returns Stringified credential
+   */
+  public async get(cid: string): Promise<string> {
+    let claim: string;
+    const getFromCluster = this.didStoreCluster.get(cid);
+    const getFromInfura = this.didStoreInfura.get(cid);
+    try {
+      claim = await Promise.race([getFromCluster, getFromInfura]);
+    } catch (e) {
+      try {
```

Comment:
If `try` failed because of getting from cluster, try to get from Infura

Reply:
Could this be simplified by using [Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any)? With this, I think this second `try` wouldn't be needed.

Reply:
Thanks. I didn't know about this
It is also requires ES2021 typescript lib